### PR TITLE
Modify MSteams secrets: add tests, add pattern readability

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml [hunk 1]
```diff
@@ -15,21 +15,32 @@ provider:
...
+        matching:
+          # Structure:
+          # https://tenant-name.webhook.office.com/webhookb2/guid@guid/IncomingWebhook/connectorId/guid
+          # An MS GUID is in the following format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX where X is a hex digit
+            pattern: "((?:https://)?\
+                    [a-z0-9_-]{1,50}\\.webhook\\.office\\.com/webhookb2/\
+                    [a-z0-9\\-]{1,50}@[a-z0-9\\-]{1,50}/IncomingWebhook/[a-z0-9]{1,50}/[a-z0-9\\-]{1,50})"
       examples:
+        - text: |
+            # Noncompliant code example
```
Comment (daniel-teuchert-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml:37]: The noncompliant code example should raise an issue. Maybe the example needs to be adjusted.

**Review by daniel-teuchert-sonarsource — CHANGES_REQUESTED** [review]
The noncompliant code example needs to be adjusted.

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml [hunk 2]
```diff
@@ -15,21 +15,32 @@ provider:
...
+            props.set("teams_webhook_url", "https://example.webhook.office.com/webhook/12345678...@.../IncomingWebhook/1bed32/39458267-2341-8657-9876-34fd6c2e5ba1")
```
Comment (loris-s-sonarsource) [inline sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml:38]:
```suggestion
            props.set("teams_webhook_url", "https://sonarcompany.webhook.office.com/webhookb2/52feb105-fe74-52b9-8e90-5d165916fe22@61c6aa5a3-6531-4e28-9c0b-33ba1a8aa1ff/IncomingWebhook/f7fb2308e5f14431ace5b7cd0e670e42/4563618c-b03b-4e80-b093-28bb4ff11de8")

```

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml (hunk 1 above)
Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/ms-teams.yaml:37]: done

**Review by daniel-teuchert-sonarsource — CHANGES_REQUESTED** [review]
The test with the noncompliant code example fails which is expected because `containsSecret` is incorrectly set to false.

**Review by daniel-teuchert-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**